### PR TITLE
Xiaomi sunsetting

### PR DESCRIPTION
**Pull request by CTLalit**
Removal of xiaomi code from the app.
Decision taken because xiaomi will no longer support push services outside of mainland china

Jira: https://wizrocket.atlassian.net/browse/SDK-3668

### CODE REVIEW

**Comment by Anush-Shand** [comment]
Proguard rule for xiaomi in sample app's proguard-rules can also be removed

File: clevertap-core/src/main/java/com/clevertap/android/sdk/utils/PackageUtils.java [hunk 1]
```diff
@@ -53,38 +51,4 @@ private static boolean isIntentResolved(Context context, Intent intent) {
...
-            String miui = (String) get.invoke(c, "ro.miui.ui.version.code");
-            if (miui!=null && !TextUtils.isEmpty(miui.trim()))
-            {
-                return true;
-            }
-        } catch (Throwable t) {
-            t.printStackTrace();
-        }
-
-        return isIntentResolved(context,
```
Comment (piyush-kukadiya) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/utils/PackageUtils.java:81]: @CTLalit we can remove `isIntentResolved ()` also as it's used by `isXiaomiDeviceRunningMiui` only.

File: clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java [hunk 2]
```diff
@@ -605,24 +601,6 @@ private List<CTPushProvider> createProviders() {
...
-            // only for XPS, if for disabled push cached token already exists then unregister xiaomi push
-            // for case like user enables xps on all devices first then in next app version disables on all devices
-
-            if (pushType == PushType.XPS) {
-                String cachedTokenXps = getCachedToken(PushType.XPS);
-                if (!TextUtils.isEmpty(cachedTokenXps)) {
-                    CTPushProvider pushProvider = getCTPushProviderFromPushType(pushType, false);
-
-                    if (pushProvider instanceof UnregistrableCTPushProvider) {
-                        ((UnregistrableCTPushProvider) pushProvider).unregisterPush(context);
```
Comment (piyush-kukadiya) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java:618]: @CTLalit We can delete `UnregistrableCTPushProvider` file

File: build.gradle [hunk 3]
```diff
@@ -55,7 +55,6 @@ task copyTemplates {
     props.put("clevertap_rendermax_sdk", libs.version.clevertap.rendermax.sdk.get())
     props.put("clevertap_geofence_sdk", libs.version.clevertap.geofence.sdk.get())
     props.put("clevertap_hms_sdk", libs.version.clevertap.hms.sdk.get())
-    props.put("clevertap_xiaomi_sdk", libs.version.clevertap.xiaomi.sdk.get())
```
Comment (piyush-kukadiya) [inline build.gradle:58]: @CTLalit we should remove xps related code from build gradle task copyTemplates

**Review by piyush-kukadiya — CHANGES_REQUESTED** [review]
Added comments

File: clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java (hunk 2 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java:594]: done

File: clevertap-core/src/main/java/com/clevertap/android/sdk/utils/PackageUtils.java (hunk 1 above)
Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/utils/PackageUtils.java:81]: done

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

**Pull request merged**